internal/ramses/reader.go

```diff
@@ -0,0 +1,58 @@
+package ramses
+
+import (
+	"encoding/binary"
+	"io"
+
+	"github.com/pkg/errors"
+)
+
+// Package is complete Ramses package
+type Package struct {
+	Header       Ramses
+	SecureHeader Secure
+	Payload      []byte
+}
+
+//Packages reads Ramses packages from buffer
+func Packages(buf io.Reader, packageChannel chan<- Package, errorChannel chan<- error) {
+	defer close(packageChannel)
+	defer close(errorChannel)
+	var err error
+	for {
+		header := Ramses{}
+		err = header.Read(buf)
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			errorChannel <- err
+			break
+		}
+
+		if !header.Valid() {
+			errorChannel <- errors.New("Not a valid RAC-file")
+			break
+		}
+
+		secureHeader := Secure{}
+		if header.SecureTrans() {
```

Comment:
this test, including the error handling seems not tested. And again, it seems more useful to have the filename/key, the alternative is ofc if the consumer of the error channel adds it.

Reply:
test reading payload with secure header.